allofplos/transformations.py

```diff
@@ -90,12 +90,12 @@ def filename_to_doi(filename):
     Uses regex to make sure it's a file and not a DOI
     Example:
     filename_to_doi('journal.pone.1000001.xml') = '10.1371/journal.pone.1000001'
-    
+
     :param filename: relative path to local XML file in the get_corpus_dir() directory
     :return: full unique identifier for a PLOS article
     """
     if correction in filename and validate_filename(filename):
-        article = 'annotation/' + (filename.split('.', 4)[2])
+        article = 'annotation/' + ((os.path.basename(filename)).split('.', 4)[2])
```

Comment:
Why not do `filename = os.path.basename(filename)` before `if correction in filename and validate_filename(filename):`?